core/Symbols.cc

```diff
@@ -604,6 +604,25 @@ bool Symbol::isHiddenFromPrinting(const GlobalState &gs) const {
     return false;
 }
 
+bool Symbol::isPrintable(const GlobalState &gs) const {
+    if (!this->isHiddenFromPrinting(gs)) {
+        return true;
+    }
+
+    for (auto childPair : this->members()) {
+        if (childPair.first == Names::singleton() || childPair.first == Names::attached() ||
+            childPair.first == Names::classMethods()) {
```

Comment:
Why are these names special cased?

Reply:
I'm guessing that depending on the motivation, these are not the only methods that are important, that there are actually more currently, and that there will be more added in the future, so I'm curious to hear why so I can suggest a potentially more future-proof implementation.

Reply:
They were special cased in the current implementation: https://github.com/sorbet/sorbet/pull/3398/files#diff-2fadf5db34eeb7dc0558a699bb2e8860L762-L765 (Similarly in `proto.cc`) 

Apparently these names are not meant to be printed, copying that logic here ensures we don't mark them as `printable`.

Reply:
Do we have tools relying on that? For the sake of the developer, it might be better to always show them and let the tool ignore what should be ignored.

Reply:
@KaanOzkan your link doesn't go anywhere for me

Reply:
Sorry I shouldn't have used a PR link. Here you go: https://github.com/sorbet/sorbet/blob/master/core/Symbols.cc#L762-L765

Reply:
I found the relevant lines.

Looking at those names now, those lines specifically are symbols that store internal stuff (like how we implement attachedClass).

Looking at those now, I think it's fine to not print those, and I'm not too worried about extra stuff ending up in the output.